### PR TITLE
feat: implement rest parameters validation

### PR DESCRIPTION
**Pull request by hsiliev**
Validates that REST parameters are present in SmartDocuments templates and groups

Solves: PZ-2779

### CODE REVIEW

File: src/itest/kotlin/nl/lifely/zac/itest/SmartDocumentsTest.kt [hunk 1]
```diff
@@ -62,51 +64,51 @@ class SmartDocumentsTest : BehaviorSpec({
...
 
-        When("the create mapping endpoint is called") {
+        When("the create mapping endpoint is called with correct payload") {
             val smartDocumentsZaakafhandelParametersUrl =
                 "$ZAC_API_URI/smartdocuments/templates/zaakafhandelParamaters/$ZAAKTYPE_MELDING_KLEIN_EVENEMENT_UUID"
             val restTemplateGroups = """
                 [
                   {
-                    "id": "6e0658a4-3039-4a7c-84d2-4ef47285ea52",
+                    "id": "D5037631FF6748269059B353699EFA0C",
```
Comment (edgarvonk) [inline src/itest/kotlin/nl/lifely/zac/itest/SmartDocumentsTest.kt:73]: Why not create constants out of these 'UUID's? Some of them are used more than once so I think it is more readable & maintainable to make constants out of them and put them in `ItestConfiguration`?

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsService.kt [hunk 2]
```diff
@@ -115,3 +116,21 @@ class SmartDocumentsService @Inject constructor(
         ).resultList.toSet().toREST()
     }
 }
+
+/**
+ * Validates that all entries in a REST request are part of pre-defined superset of groups and templates.
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsService.kt:121]: Maybe I am overlooking things but where is the REST request you mention here in this function? From reading this I would expect a function param for the REST request?

File: src/main/kotlin/net/atos/zac/smartdocuments/templates/SmartDocumentsTemplateConverter.kt [hunk 3]
```diff
@@ -127,4 +127,24 @@ object SmartDocumentsTemplateConverter {
...
+
+    private fun convertTemplateGroupToStringRepresentation(
+        group: RESTSmartDocumentsTemplateGroup,
+        parent: String?
+    ): Set<String> {
+        val thisGroupString = "group.${group.id}.${group.name}"
+        val groupString = if (parent.isNullOrEmpty()) thisGroupString else "$parent.$thisGroupString"
+        val groupRepresentation = mutableSetOf(groupString)
+
+        group.templates?.mapTo(groupRepresentation) { "$groupString.template.${it.id}.${it.name}" }
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/templates/SmartDocumentsTemplateConverter.kt:145]: A little bit hard to read this code. I think we can probably make this a bit more functional / Kotlin idiomatic. @bas-info-nl is usually very good at that kind of stuff. ;-)

**Review by edgarvonk — CHANGES_REQUESTED** [review]

File: src/itest/kotlin/nl/lifely/zac/itest/SmartDocumentsTest.kt (hunk 1 above)
Comment (hsiliev) [reply src/itest/kotlin/nl/lifely/zac/itest/SmartDocumentsTest.kt]: Moved the IDs and names to ITestConfiguration

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsService.kt (hunk 2 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsService.kt]: Changed the JavaDoc to clarify that we validate RESTSmartDocumentsTemplateGroup against another RESTSmartDocumentsTemplateGroup

**Review by edgarvonk — APPROVED** [review]

**Pull request merged**